perf-event/src/events/msr.rs

```diff
@@ -0,0 +1,68 @@
+use crate::events::Event;
+use c_enum::c_enum;
+use perf_event_open_sys::bindings;
+use std::io;
+use std::sync::atomic::{AtomicU32, Ordering};
+
+// 0 will never be the PMU value for msr.
+// We use it as a flag value to indicate that this has not been initialized.
+static MSR_TYPE: AtomicU32 = AtomicU32::new(0);
+
+
+c_enum! {
+    /// MSR events on all x86 CPUs
+    #[derive(Clone, Copy, Eq, PartialEq, Hash)]
+    pub enum MSRConfig: u64 {
+        /// Time Stamp Counter
+        /// TSC increases at the CPU base frequency.
+        TSC = 0x0,
+        /// Actual Performance Frequency Clock Count
+        APERF = 0x1,
+        /// Maximum Performance Frequency Clock Count
+        /// (APERF / MPERF) * CPU_BASE_FREQUENCY is the running CPU frequency
+        MPERF = 0x2,
+    }
+}
+
+/// MSR event
+/// 
+/// MSR event allow you to read the per-CPU x86 MSR counters. The MSR event might not exist on non-x86 CPUs
+/// and the type of the pmu is dynamic, so MSREvent::with_config searches /sys/fs/event_source/devices/msr/type
+/// checking whether it exists and fetching the run-time type value.
+pub struct MSREvent {
```

Comment:
This is in the events module. I don't think the type needs to have `Event` in the name
```suggestion
pub struct Msr {
```